indico/modules/oauth/controllers.py

```diff
@@ -16,14 +16,137 @@
 
 from __future__ import unicode_literals
 
-from indico.core.index import Catalog
+from flask import flash, redirect, request, render_template, session
+from werkzeug.exceptions import Forbidden
+
+from indico.core.db import db
 from indico.modules.users.controllers import RHUserBase
-from indico.modules.oauth.views import WPOAuthUserProfile
+from indico.modules.oauth import logger
+from indico.modules.oauth.provider import oauth
+from indico.modules.oauth.forms import ApplicationForm
+from indico.modules.oauth.models.applications import OAuthApplication
+from indico.modules.oauth.models.tokens import OAuthToken
+from indico.modules.oauth.views import WPOAuthUserProfile, WPOAuthAdmin
+from indico.util.i18n import _
+from indico.web.flask.util import url_for
+from MaKaC.webinterface.rh.admins import RHAdminBase
+from MaKaC.webinterface.rh.base import RH, RHProtected
+
+
+class RHOAuthAuthorize(RHProtected):
+    def _checkParams(self):
+        self.application = OAuthApplication.find_one(client_id=request.args['client_id'])
+
+    @oauth.authorize_handler
+    def _process(self, **kwargs):
+        if request.method == 'GET':
+            if self.application.is_trusted:
+                return True
+            return render_template('oauth/authorize.html', application=self.application)
+        confirm = request.form.get('confirm', 'no')
+        return confirm == 'yes'
+
+
+class RHOAuthToken(RH):
+    @oauth.token_handler
```

Comment:
If this always returns none, it's rather pointless to have it in a RH. A plain function would do the job just fine.